src/chat/chatroom.js

```diff
@@ -21,45 +23,63 @@ export default class Chatroom extends Component {
     this.state = {
       message: null,
       messageList: [],
-      location: '37.7837-122.4090',
+      location: '37.784-122.409',
       demoMode: true,
       userLoggedIn: false,
+      username: 'Hannah Test Person'
     }
 
+    this.emitAddMessageToChatRoom();
+
   }
 
   componentWillMount() {
-    this.onSendPress = this.onSendPress.bind(this);
-    this.onGetChat = this.onGetChat.bind(this)
+    this.getMessagesOnMount = this.getMessagesOnMount.bind(this);
+    this.getMessagesOnMount();
   }
 
-  onGetChat() {
-    this.props.socket.on('updateMessagesState', updatedChatRoom => {
-      console.log('messages received:', updatedChatRoom);
-      this.setState({ messageList: updatedChatRoom });
-    })
+  onSendPress() {
+    this.emitAddMessageToChatRoom()
+    this.setState({ message: ''})
   }
 
-  onSendPress() {
+  emitAddMessageToChatRoom() {
     this.props.socket.emit('addMessageToChatRoom', { 
         location: this.state.location,
         message: this.state.message,
-        username: null // TODO: Add Username
+        username: this.state.username // TODO: Add Username
       });
+  }
 
-    console.log('addMessageToChatRoom:', this.state.message);
-
-    this.setState({
-      message: '',
+  getMessagesOnMount() {
+    this.props.socket.on('updateMessagesState', updatedChatRoom => {
+      console.log('messages received:', updatedChatRoom.messages);
+      const messages = updatedChatRoom ? updatedChatRoom.messages : null;
+      this.setState({ messageList: messages });
     })
-
   }
 
   onBackPress() {
     console.log('navigate back a page');
   }
 
   render() {
+    var list = this.state.messageList.map((item, index) => {
```

Comment:
Looks like this would be a good candidate for another component. Maybe add a TODO